### PR TITLE
[h264d] Enable the SPS/PPS error reporting

### PR DESCRIPTION
**Pull request by UngTengEn**
Issue:N/A
Test:sample_decode with mfxExtDecodeErrorReport

Signed-off-by: Ung, Teng En <teng.en.ung@intel.com>

### CODE REVIEW

File: _studio/mfx_lib/decode/h265/src/mfx_h265_dec_decode.cpp [hunk 1]
```diff
@@ -1084,6 +1084,16 @@ mfxStatus VideoDECODEH265::DecodeFrameCheck(mfxBitstream *bs, mfxFrameSurface1 *
 
         MFXMediaDataAdapter src(bs);
 
+#if (MFX_VERSION >= 1025)
+        mfxExtBuffer* extbuf = (bs) ? GetExtendedBuffer(bs->ExtParam, bs->NumExtParam, MFX_EXTBUFF_DECODE_ERROR_REPORT) : NULL;
+
+        if (extbuf)
+        {
+            ((mfxExtDecodeErrorReport *)extbuf)->ErrorTypes = 0;
```
Comment (mgonchar) [inline _studio/mfx_lib/decode/h265/src/mfx_h265_dec_decode.cpp:1092]: Here we actually change data on user side, not sure that it is a good idea

File: _studio/shared/umc/codec/h265_dec/src/umc_h265_task_supplier.cpp [hunk 2]
```diff
@@ -47,6 +47,18 @@
 namespace UMC_HEVC_DECODER
 {
 
+#if (MFX_VERSION >= 1025)
+inline void SetDecodeErrorTypes(NalUnitType nalUnit, mfxExtDecodeErrorReport *pDecodeErrorReport)
```
Comment (mgonchar) [inline _studio/shared/umc/codec/h265_dec/src/umc_h265_task_supplier.cpp:51]: I'd suggest to accept ```mfxExtDecodeErrorReport``` as reference, or you have to check pointer for null

File: _studio/shared/umc/codec/h265_dec/src/umc_h265_task_supplier.cpp [hunk 3]
```diff
@@ -1638,6 +1650,11 @@ UMC::Status TaskSupplier_H265::ProcessNalUnit(UMC::MediaDataEx *nalUnit)
     UMC::MediaDataEx::_MediaDataEx* pMediaDataEx = nalUnit->GetExData();
     NalUnitType unitType = (NalUnitType)pMediaDataEx->values[pMediaDataEx->index];
 
+#if (MFX_VERSION >= 1025)
```
Comment (mgonchar) [inline _studio/shared/umc/codec/h265_dec/src/umc_h265_task_supplier.cpp:1653]: as I understand, we need this code only if we get to line 1682. So let's put it there

File: _studio/shared/umc/codec/h265_dec/src/umc_h265_task_supplier.cpp [hunk 4]
```diff
@@ -1703,6 +1725,11 @@ UMC::Status TaskSupplier_H265::AddOneFrame(UMC::MediaData * pSource)
 
         UMC::MediaDataEx::_MediaDataEx* pMediaDataEx = nalUnit->GetExData();
 
+#if (MFX_VERSION >= 1025)
+    UMC::MediaData::AuxInfo* aux = (pSource) ? pSource->GetAuxInfo(MFX_EXTBUFF_DECODE_ERROR_REPORT) : NULL;
```
Comment (mgonchar) [inline _studio/shared/umc/codec/h265_dec/src/umc_h265_task_supplier.cpp:1729]: code alignment here doesn't match to previous lines

File: _studio/mfx_lib/decode/h265/src/mfx_h265_dec_decode.cpp (hunk 1 above)
Comment (UngTengEn) [reply _studio/mfx_lib/decode/h265/src/mfx_h265_dec_decode.cpp:1092]: We report back the ErrorType to user.  Should be fine to clear it before we use it first time.  Just want to make sure there is no previous left value inside.

File: _studio/shared/umc/codec/h265_dec/src/umc_h265_task_supplier.cpp (hunk 2 above)
Comment (UngTengEn) [reply _studio/shared/umc/codec/h265_dec/src/umc_h265_task_supplier.cpp:51]: Will submit a new patch on this.

File: _studio/shared/umc/codec/h265_dec/src/umc_h265_task_supplier.cpp (hunk 3 above)
Comment (UngTengEn) [reply _studio/shared/umc/codec/h265_dec/src/umc_h265_task_supplier.cpp:1653]: Will submit a new patch on this.

File: _studio/shared/umc/codec/h265_dec/src/umc_h265_task_supplier.cpp (hunk 4 above)
Comment (UngTengEn) [reply _studio/shared/umc/codec/h265_dec/src/umc_h265_task_supplier.cpp:1729]: Ok, will align that.

File: _studio/shared/umc/codec/h265_dec/src/umc_h265_task_supplier.cpp (hunk 2 above)
Comment (UngTengEn) [reply _studio/shared/umc/codec/h265_dec/src/umc_h265_task_supplier.cpp:51]: I am new in GitHub MediaSDK, new patches on reviewed commit should submit as new commits or merge with original commit?

File: _studio/mfx_lib/decode/h265/src/mfx_h265_dec_decode.cpp (hunk 1 above)
Comment (mgonchar) [reply _studio/mfx_lib/decode/h265/src/mfx_h265_dec_decode.cpp:1092]: oh, ok

**Comment by mgonchar** [comment]
@UngTengEn you've uploaded merge commit, please, rebase your change and upload just your new commit with force-push. WE should have clean history before submitting the patch

File: _studio/shared/umc/codec/h265_dec/src/umc_h265_task_supplier.cpp (hunk 2 above)
Comment (mgonchar) [reply _studio/shared/umc/codec/h265_dec/src/umc_h265_task_supplier.cpp:51]: @UngTengEn you need to amend your commit localy and force push it, so there should be only one single commit in history

File: _studio/shared/umc/codec/h265_dec/src/umc_h265_task_supplier.cpp [hunk 5]
```diff
@@ -1662,6 +1677,13 @@ UMC::Status TaskSupplier_H265::ProcessNalUnit(UMC::MediaDataEx *nalUnit)
     case NAL_UT_SPS:
     case NAL_UT_PPS:
         umcRes = DecodeHeaders(nalUnit);
+#if (MFX_VERSION >= 1025)
+        UMC::MediaData::AuxInfo* aux = (nalUnit) ? nalUnit->GetAuxInfo(MFX_EXTBUFF_DECODE_ERROR_REPORT) : NULL;
```
Comment (mgonchar) [inline _studio/shared/umc/codec/h265_dec/src/umc_h265_task_supplier.cpp:1681]: CI is complaining on initialization inside case, to fix that just introduce new block:
```
{
    #if (MFX_VERSION >= 1025)
        UMC::MediaData::AuxInfo* aux = (nalUnit) ? nalUnit->GetAuxInfo(MFX_EXTBUFF_DECODE_ERROR_REPORT) : NULL;
...
}
```

Comment (UngTengEn) [reply _studio/shared/umc/codec/h265_dec/src/umc_h265_task_supplier.cpp]: Added the block in new patch.

**Review by onabiull — CHANGES_REQUESTED** [review]
Please rebase.

**Review by lakulako — COMMENTED** [review]
Please rebase.

**Comment by UngTengEn** [comment]
> Please rebase.

I already rebase on UngTengEn:dev_hevc_report_error.  Is tere anything I need to do there?

**Comment by UngTengEn** [comment]
> Please rebase.

I already rebase on  UngTengEn:dev_hevc_report_error.  Is there anything I need to do?

**Comment by lakulako** [comment]
I see 159 files changed in your commit. Most changes are from already merged commits. Probably you should update your repo with MediaSDK:master first, then rebase the commit on updated repo?

**Pull request closed without merging**